docs/design/mp_pod_to_vpc_subnet.md

```diff
@@ -0,0 +1,126 @@
+<!--
+SPDX-License-Identifier: MIT
+Copyright (c) 2021 The Authors.
+
+Authors: Phu Tran          <@phudtran>
+
+-->
+
+# Mizar User defined VPCs and Subnets for Vanilla Kubernetes
+
+## Introduction
+
+This document defines the interface for users to define their own VPCs and subnets. Furthermore, it also describes the interface for assigning pods to these specific subnets and VPCs.
+
+## Custom VPC Creation
+
+When creating a VPC, users must define the VPC's name, IP, prefix, and the number of dividers.
+
+An example is given below.
+
+```yaml
+apiVersion: "mizar.com/v1"
+kind: Vpc
+metadata:
+  name: vpc0
+spec:
+  ip: "12.0.0.0"
```

Comment:
I found there is no vni in subnet and vpc.  Do we still need vni or vni is not user defined any more?